### ISSUE TITLE
Catastrophic failure sets Max R&D 1 point too low

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#545]
I've noticed that occasionally a catastrophic failure sets a component's Max R&D is set slightly too low after a catastrophic failure.  Today for instance, my R-7 blew up, taking a cosmonaut with it, and the Max R&D was set to 84 instead of 85.  This may be related to #282, which seems to be a similar issue.
![R-7MaxR D](https://user-images.githubusercontent.com/3118014/120899372-45407680-c5e4-11eb-9713-d988896a9500.png)

**Comment by peyre** [comment]
And here's Gemini after my crew smashed into the Moon, craft out of control.  Should be 83% Max R&D, but it's down to 82.

![Gemini82](https://user-images.githubusercontent.com/3118014/122692053-d2d8b480-d1e7-11eb-9626-a47ebb52ffdb.png)

**Comment by peyre** [comment]
Could the culprit be

`    GetEquipment(Mev[STEP])->MaxRD = GetEquipment(Mev[STEP])->MSF - 1;`

in mis_m.cpp?  I might have to try playing with that.

**Comment by peyre** [comment]
Hum!  I took off the -1, recompiled, and I just suffered a catastrophic failure on the Vostok - and now I have the opposite problem.  It reset Max R&D to 77% instead of 76 (or 75).  Maybe what we should do is store the original Max R&D for everything in a discrete set of variables so there's no need to do weird stuff with -1 or whatever.

**Comment by peyre** [comment]
Yes, it's inconsistent.  I'm on another game, flying Soyuz - which has suffered two catastrophic failures.  The first time it set the Max R&D to 85% like it should have, but the second time it set it to 86%.  I wonder why it does that!

**Comment by rnyoakum** [comment]
> Yes, it's inconsistent. I'm on another game, flying Soyuz - which has suffered two catastrophic failures. The first time it set the Max R&D to 85% like it should have, but the second time it set it to 86%. I wonder why it does that!

I'll try and take a look at this, but here's my guess: it depends on when in the mission the catastrophic failure occurs. One of the benefits of test flights is that the Max R&D value of the hardware used increases if the craft succeeds at a test during the mission. The craft doesn't have to succeed at every test to gain this advantage, because that's unreasonable. So if the craft has succeeded at a mission step before the catastrophic failure ends the mission, it's given the +1 Max R&D bonus as expected. Consequently
```
GetEquipment(Mev[STEP])->MaxRD = GetEquipment(Mev[STEP])->MSF - 1;
```
attempts to compensate for the bonus which will be received at the end of the mission, but _if the hardware hasn't succeeded thus far the bonus won't be given._ This leaves MaxRD one point low.

**Comment by peyre** [comment]
Ohhhh!  That would explain it all right.  My goodness, how do you figure these things out?  It would be nice to get this one straight, since the Max R&D really shouldn't be set above the original but it feels cruel (esp. for the Soviets) for it to be set lower than original.

**Comment by peyre** [comment]
Ok, until we can fix this, I think I'll take off the -1.
